pkg/types/vector.go

```diff
@@ -93,7 +100,60 @@ func (v VectorFloat32) Elements() []float32 {
 	return unsafe.Slice((*float32)(unsafe.Pointer(&v.data[4])), l)
 }
 
-// String returns a string representation of the vector, which can be parsed later.
+// StringWithCtx implements Explainable interface.
+// In EXPLAIN context, we truncate the elements to avoid too long output.
+func (v VectorFloat32) StringWithCtx(ctx ParamValues, redact string) string {
+	return v.StringWithRedact(ctx, redact)
+}
+
+// StringWithRedact parse vector into string with redact mode.
+func (v VectorFloat32) StringWithRedact(ctx ParamValues, redact string) string {
```

Comment:
Ctx and redact is not used at all. From Vector's perspective you should remove them.